darwin/Classes/AudioPlayer.m

```diff
@@ -335,7 +335,9 @@ - (int)getCurrentPosition {
     } else if (CMTIME_IS_VALID(_seekPos)) {
         return (int)(1000 * CMTimeGetSeconds(_seekPos));
     } else if (_indexedAudioSources && _indexedAudioSources.count > 0) {
-        int ms = (int)(1000 * CMTimeGetSeconds(_indexedAudioSources[_index].position));
+        CMTimeRange seekableRange = [_indexedAudioSources[_index].playerItem.seekableTimeRanges.lastObject CMTimeRangeValue];
+        CGFloat seekableStart = CMTimeGetSeconds(seekableRange.start);
+        int ms = (int)(1000 * CMTimeGetSeconds(_indexedAudioSources[_index].position)) - (seekableStart * 1000);
```

Comment:
I think this needs to be encapsulated inside of the relevant indexed audio source. In other words, leave this line as per the original:

```dart
int ms = (int)(1000 * CMTimeGetSeconds(_indexedAudioSources[_index].position))
```

but then allow the implementation of the `position` property to figure out the correct position, since even the seekable time ranges will be offset for a `ClippingAudioSource`. All you would need to do is push this code into `UriAudioSource`. If you look at the implementation of `ClippingAudioSource` you can see that it will automatically benefit from this code, and then correctly offset the position.

The same applies to `duration` and `seek` which are also encapsulated within the relevant indexed audio source because the behaviour of each of these is dependent on what type of indexed audio source it is.